NPE in GenericThingHandler

```
2017-06-11 16:53:42.430 [ERROR] [.core.internal.folder.FolderObserver] - Error handling update of file '/openhab/conf/things/sjka.things': null.
java.lang.NullPointerException
	at org.eclipse.smarthome.model.thing.internal.GenericThingProvider.getAllThingUIDs(GenericThingProvider.java:870)[134:org.eclipse.smarthome.model.thing:0.9.0.201706091022]
	at org.eclipse.smarthome.model.thing.internal.GenericThingProvider.modelChanged(GenericThingProvider.java:834)[134:org.eclipse.smarthome.model.thing:0.9.0.201706091022]
	at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.notifyListeners(ModelRepositoryImpl.java:286)[123:org.eclipse.smarthome.model.core:0.9.0.201706091022]
	at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.addOrRefreshModel(ModelRepositoryImpl.java:136)[123:org.eclipse.smarthome.model.core:0.9.0.201706091022]
	at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.checkFile(FolderObserver.java:234)[123:org.eclipse.smarthome.model.core:0.9.0.201706091022]
	at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.processWatchEvent(FolderObserver.java:297)[123:org.eclipse.smarthome.model.core:0.9.0.201706091022]
	at org.eclipse.smarthome.core.service.WatchQueueReader.run(WatchQueueReader.java:206)[99:org.eclipse.smarthome.core:0.9.0.201706091022]
	at java.lang.Thread.run(Thread.java:745)[:1.8.0_112]
```

What I did:
* Updated a binding which now requires different configuration parameters (with my old config it failed with NPEs, just in case it matters)
* I corrected them (i.e. added the missing config)
* I renamed by things within the bridge (as with the new config I am allowed to freely chose their names)
* Ended up with old AND new things, all with "UNKNOWN"  status

![image](https://user-images.githubusercontent.com/2592443/27011943-14f442ce-4ec7-11e7-9053-5dedf9633317.png)

Reboot helped.

--- a/bundles/model/org.eclipse.smarthome.model.thing/src/org/eclipse/smarthome/model/thing/internal/GenericThingProvider.xtend
+++ b/bundles/model/org.eclipse.smarthome.model.thing/src/org/eclipse/smarthome/model/thing/internal/GenericThingProvider.xtend
@@ -444,13 +444,14 @@ class GenericThingProvider extends AbstractProvider<Thing> implements ThingProvi
                 case org.eclipse.smarthome.model.core.EventType.MODIFIED: {
                     val oldThings = thingsMap.get(modelName) ?: newArrayList
                     val model = modelRepository.getModel(modelName) as ThingModel
-                    val newThingUIDs = model.allThingUIDs
-                    val removedThings = oldThings.filter[!newThingUIDs.contains(it.UID)]
-                    removedThings.forEach [
-                        notifyListenersAboutRemovedElement
-                    ]
-
-                    createThingsFromModel(modelName)
+                    if (model != null) {
+                        val newThingUIDs = model.allThingUIDs
+                        val removedThings = oldThings.filter[!newThingUIDs.contains(it.UID)]
+                        removedThings.forEach [
+                            notifyListenersAboutRemovedElement
+                        ]
+                        createThingsFromModel(modelName)
+                    }
                 }
                 case org.eclipse.smarthome.model.core.EventType.REMOVED: {
                     val things = thingsMap.remove(modelName) ?: newArrayList

--- a/bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/GenericItemUIProvider.java
+++ b/bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/GenericItemUIProvider.java
@@ -65,9 +65,12 @@ public class GenericItemUIProvider implements ItemUIProvider {
         if (itemName != null && modelRepository != null) {
             for (String modelName : modelRepository.getAllModelNamesOfType("items")) {
                 ItemModel model = (ItemModel) modelRepository.getModel(modelName);
-                for (ModelItem item : model.getItems()) {
-                    if (itemName.equals(item.getName()))
-                        return item;
+                if (model != null) {
+                    for (ModelItem item : model.getItems()) {
+                        if (itemName.equals(item.getName())) {
+                            return item;
+                        }
+                    }
                 }
             }
         }

